trinity/extensibility/plugin.py

```diff
@@ -6,17 +6,43 @@
     ArgumentParser,
     _SubParsersAction,
 )
+from enum import (
+    auto,
+    Enum,
+)
 import logging
 from typing import (
-    Any
+    Any,
+    Dict,
+)
+
+from lahja import (
+    Endpoint
 )
 
 from trinity.extensibility.events import (
     BaseEvent
 )
+from trinity.utils.mp import (
+    ctx,
+)
+from trinity.utils.logging import (
+    setup_queue_logging,
+)
+
+
+class PluginProcessScope(Enum):
 
-# TODO: we spec this out later
-PluginContext = Any
+    SHARED = auto()
+    MAIN = auto()
+    OWN = auto()
```

Comment:
Maybe 

-`PRIVATE` 
-`ISOLATED`

Also, these probably deserve comments indicating what they mean here in the code.

Reply:
Agreed. I like `ISOLATED` for plugins that run in their own process. `SHARED` means, the plugin runs in a process that is shared among other plugins (currently `networking` but long term I see them running in a shared process that is exclusively for plugins)